### PR TITLE
[FLINK-22021][table-planner-blink] Fix exception when PushFilterIntoLegacyTableSourceScanRule is faced with INTERVAL types

### PR DESCRIPTION
**Pull request by TsReaper**
## What is the purpose of the change

`PushFilterIntoLegacyTableSourceScanRule` will throw exception when faced with `INTERVAL` types, no matter it can be pushed down or not.

This PR fixes this issue.

## Brief change log

 - Fix exception when PushFilterIntoLegacyTableSourceScanRule is faced with INTERVAL types

## Verifying this change

This change can be verified by running the added tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PushFilterIntoLegacyTableSourceScanRuleTest.xml [hunk 1]
```diff
@@ -287,27 +333,6 @@ LogicalProject(name=[$0], id=[$1], amount=[$2], price=[$3])
 LogicalProject(name=[$0], id=[$1], amount=[$2], price=[$3])
 +- LogicalFilter(condition=[<(org$apache$flink$table$planner$expressions$utils$Func1$$879c8537562dbe74f3349fa0e6502755($2), 32)])
    +- LogicalTableScan(table=[[default_catalog, default_database, MyTable, source: [filterPushedDown=[true], filter=[greaterThan(amount, 2)]]]])
-]]>
-    </Resource>
-  </TestCase>
-  <TestCase name="testPartialPushDownWithVirtualColumn">
```
Comment (JingsongLi) [inline flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PushFilterIntoLegacyTableSourceScanRuleTest.xml:293]: Can you keep this as it is?

File: flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PushFilterIntoTableSourceScanRuleTest.xml [hunk 2]
```diff
@@ -287,27 +333,6 @@ LogicalProject(name=[$0], id=[$1], amount=[$2], price=[$3])
 LogicalProject(name=[$0], id=[$1], amount=[$2], price=[$3])
 +- LogicalFilter(condition=[<(org$apache$flink$table$planner$expressions$utils$Func1$$879c8537562dbe74f3349fa0e6502755($2), 32)])
    +- LogicalTableScan(table=[[default_catalog, default_database, MyTable, filter=[greaterThan(amount, 2)]]])
-]]>
-    </Resource>
-  </TestCase>
-  <TestCase name="testPartialPushDownWithVirtualColumn">
```
Comment (JingsongLi) [inline flink-table/flink-table-planner-blink/src/test/resources/org/apache/flink/table/planner/plan/rules/logical/PushFilterIntoTableSourceScanRuleTest.xml:293]: Can you keep this as it is?

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala [hunk 3]
```diff
@@ -408,6 +408,16 @@ class RexNodeToExpressionConverter(
...
 
+      case INTERVAL_DAY_TIME =>
+        // milliseconds
+        val v = literal.getValueAs(classOf[java.lang.Long])
+        java.time.Duration.ofMillis(v)
+
+      case INTERVAL_YEAR_MONTH =>
+        // months
+        val v = literal.getValueAs(classOf[java.lang.Integer])
+        java.time.Period.ofMonths(v)
```
Comment (JingsongLi) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala:419]: You should take a look to `YearMonthIntervalPeriodConverter`.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
Thanks @TsReaper for your contribution, left some comments.

**Comment by TsReaper** [comment]
Azure passed in https://dev.azure.com/tsreaper96/Flink/_build/results?buildId=137&view=results

**Review by JingsongLi — APPROVED** [review]
Looks good to me.

**Pull request merged**